src/main/scala/core/crawler/Crawler.scala

```diff
@@ -65,10 +66,41 @@ object Crawler:
                                maxDepth: Int,
                                networkConfiguration: ClientConfiguration = Configuration.default
                              ): Behavior[CrawlerCommand] =
+    Crawler[T](coordinator, exporter, scrapeRule, explorationPolicy,
+      maxDepth, SimpleHttpClient(networkConfiguration))
+
+  /**
+   * Creates a new Crawler actor with an already set up [[SimpleHttpClient]].
+   *
+   * @param coordinator
+   *    the ActorRef of the coordinator to communicate with
+   * @param exporter
+   *    the ActorRef of the exporter to communicate with
+   * @param scrapeRule
+   *    scraping rule for the [[Scraper]]s spawned by this Crawler
+   * @param explorationPolicy
+   *    policy that specifies what links to explore inside the [[CrawlDocument]]
+   * @param maxDepth
+   *    max recursion depth for crawlers (maximum depth of the tree of crawlers)
+   * @param httpClient
+   *    the http client already set up
+   * @tparam T type of [[Result]]s that get exported
+   * @return
+   *    the behavior of the Crawler actor
+   */
+  def apply[T](
+              coordinator: ActorRef[CoordinatorCommand],
+              exporter: ActorRef[ExporterCommands],
+              scrapeRule: ScraperPolicy[T],
+              explorationPolicy: ExplorationPolicy,
+              maxDepth: Int,
+              httpClient: SimpleHttpClient): Behavior[CrawlerCommand] =
+    println(httpClient)
     Behaviors.withStash(50): buffer =>
       Behaviors.setup:
-        context => new Crawler[T](context, coordinator, exporter, scrapeRule, explorationPolicy,
-          maxDepth, networkConfiguration, buffer).idle()
+        context =>
+          new Crawler[T](context, coordinator, exporter, scrapeRule, explorationPolicy,
+            maxDepth, buffer)(using httpClient).idle()
```

Comment:
Why passing httpClient with `using`? It's used as a normal parameter, what's the purpose?

Reply:
It's required for the HTTP requests to use it like a given. Anyway, I agree with @giovaz94: maybe it's better to pass it like a normal parameter and then place a `given ... = httpClient` inside the class

Reply:
Thanks for suggestions, changed signature for apply method with client, updated given client as stated by @FreshMag